nomos-services/data-availability/src/backend/memory_cache.rs

```diff
@@ -0,0 +1,73 @@
+use crate::backend::{DaBackend, DaError};
+use moka::future::{Cache, CacheBuilder};
+use nomos_core::blob::Blob;
+use std::time::Duration;
+
+#[derive(Clone, Copy)]
+pub struct BlobCacheSettings {
+    max_capacity: usize,
+    evicting_period: Duration,
+}
+
+pub struct BlobCache<H, B>(Cache<H, B>);
+
+impl<B> BlobCache<B::Hash, B>
+where
+    B: Clone + Blob + Send + Sync + 'static,
+    B::Hash: Send + Sync + 'static,
+{
+    pub fn new(settings: BlobCacheSettings) -> Self {
+        let BlobCacheSettings {
+            max_capacity,
+            evicting_period,
+        } = settings;
+        let cache = CacheBuilder::new(max_capacity as u64)
+            .time_to_live(evicting_period)
+            // can we leverage this to evict really old blobs?
+            .time_to_idle(evicting_period)
```

Comment:
As discussed in https://github.com/logos-co/nomos-node/pull/376/files#r1317083701, the expectation is that the outer service retrieves pending blobs from this service, does something with those blobs, and after that, requests removing those blobs? If so, it seems it's okay to use `time_to_idle`. This will reset eviction timers whenever `pending_blobs()` is called, but I guess it's okay if it wouldn't take too long for the outer service to process and remove blobs.

Reply:
Yes. So the idea here is that. Either a blob is pending for the whole life until is added into a block. Or is removed. (that is why is using the same evicting period as the time to live). But it would make sense that if it is not included it dies as well after some time.